moveit_simple/src/moveit_simple.cpp

```diff
@@ -219,16 +217,115 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
   return getIK(pose, local_seed, joint_point, timeout);
 }
 
+bool Robot::getJointSolution(const Eigen::Affine3d &pose, const std::string& custom_tool_frame,
+                             double timeout, const std::vector<double> & seed,
+                             std::vector<double> & joint_point) const
+{
+
+  // Ideally "tool0" should be replaced with joint_group_->getEndEffectorName()
+  std::string moveit_tool_link = "tool0";
+
+  // Transform Target/Goal Point from custom tool frame to moveit_end_link
+  const Eigen::Affine3d custom_frame_goal_pose = 
```

Comment:
`transformPoseBetweenFrames` throws...should you catch it an handle the boolean return.